src/gui/qgsprojectionselectiontreewidget.cpp

```diff
@@ -795,7 +795,7 @@ void QgsProjectionSelectionTreeWidget::lstRecent_itemDoubleClicked( QTreeWidgetI
 
 void QgsProjectionSelectionTreeWidget::updateFilter()
 {
-  QString filterTxtCopy = leSearch->text();
+  QString filterTxtCopy = QRegExp::escape( leSearch->text() );
```

Comment:
Any chance this could be upgraded to `QRegularExpression` directly to have it ready for qt 6?

Reply:
I tried with
```c++
  QString filterTxtCopy = QRegularExpression::escape( leSearch->text() );
  filterTxtCopy.replace( QRegularExpression( "\\s+" ), QStringLiteral( ".*" ) );
  const QRegularExpression re( filterTxtCopy, QRegularExpression::CaseInsensitiveOption );
```
but it seems that [QRegularExpression::escape()](https://doc.qt.io/qt-5/qregularexpression.html#escape) [introduced in Qt 5.15] escapes differently than [QRegExp::escape()](https://doc.qt.io/qt-5/qregexp.html#escape)...
Moreover I'm not sure what are the required minimum Qt versions for 3.16 and 3.20+
So I think to revert to QRegExp...

Reply:
This should address the different behaviour between QRegularExpression::escape() and QRegExp::escape()
```c++
  filterTxtCopy.replace( QRegularExpression( R"(?:\\\s)+" ), QStringLiteral( ".*" ) );
```
Anyway I don't know how to deal with the required minimum Qt versions for 3.16 and 3.20+ so I revert to QRegExp.

Reply:
Fair enough